src/libs/Navigation/AppNavigator/MainDrawerNavigator.js

```diff
@@ -100,7 +104,14 @@ class MainDrawerNavigator extends Component {
         if (!this.initialParams.reportID) {
             const state = this.props.navigation.getState();
             const reportScreenKey = lodashGet(state, 'routes[0].state.routes[0].key', '');
-            Navigation.setParams(initialNextParams, reportScreenKey);
+
+            const reportScreenReportID = this.props.reportScreenReportID.split('/r/')[1];
+
+            // If the user is trying to get to a chat before logging in
+            // do not override the chat with the most recent chat.
+            if (!reportScreenReportID) {
+                Navigation.setParams(initialNextParams, reportScreenKey);
+            }
```

Comment:
```suggestion
            // If this onyx key exists then it means we stashed the url from a deep link so we want to override the reportID that is being loaded by default to the one the user was attempting to load initially.
            const reportScreenReportID = this.props.reportScreenReportID.split('/r/')[1];
            if (!_.isEmpty(reportScreenReportID)) {
                initialNextParams.reportID = reportScreenReportID;
            }
            
            Navigation.setParams(initialNextParams, !_.isEmpty(reportScreenReportID) ? reportScreenReportID : reportScreenKey);
```

I don't love my comment so open to other refinement but I tested this on web and it seems to work.